overlord/ifacestate/handlers.go

```diff
@@ -507,6 +530,83 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// doReconnect creates a set of tasks for connecting the interface and running its hooks
+func (m *InterfaceManager) doReconnect(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapsup, err := snapstate.TaskSnapSetup(task)
+	if err != nil {
+		return err
+	}
+
+	snapName := snapsup.Name()
+	connections, err := m.repo.Connections(snapName)
+	if err != nil {
+		return err
+	}
+
+	connectts := state.NewTaskSet()
+	chg := task.Change()
+	for _, conn := range connections {
+		ts, err := Connect(st, conn.PlugRef.Snap, conn.PlugRef.Name, conn.SlotRef.Snap, conn.SlotRef.Name)
+		if err != nil {
+			return err
+		}
+		connectts.AddAll(ts)
+	}
+
+	task.SetStatus(state.DoneStatus)
+
+	lanes := task.Lanes()
```

Comment:
we have this kind of code to add tasks and consider lanes in a couple of places now,  is it worth making an helper?

Reply:
That's right. If that's OK with you then I'd prefer to address this in a followup PR and fix all these occurences in same PR, it will be more visible what's happening.

Reply:
it's fine in a follow up